pkg/sql/opt/norm/fold_constants.go

```diff
@@ -66,6 +66,29 @@ func (c *CustomFuncs) IsConstValueOrTuple(input opt.ScalarExpr) bool {
 	return memo.CanExtractConstDatum(input)
 }
 
+// IsNullTuple returns true if the input tuple has all null elements.
+func (c *CustomFuncs) IsNullTuple(input opt.ScalarExpr) bool {
+	tup := input.(*memo.TupleExpr)
+	for _, e := range tup.Elems {
+		if e.Op() != opt.NullOp {
+			return false
+		}
+	}
+	return true
+}
+
+// IsNonNullTuple returns true if the input tuple has one non-null constant
+// value.
+func (c *CustomFuncs) IsNonNullTuple(input opt.ScalarExpr) bool {
```

Comment:
We currently only fold `tuple IS NULL` to `true` if `tuple` has only constants. I'm not sure why we don't fold in the case of `(1, col1) IS NULL` to `False`.

So I'm a bit scared of this change because we will now fold `(1, col1) IS NULL` to `False`. Is there a reason not to do this folding?